### DISCUSSION TITLE
Valor padrão em atributo de Class

### DISCUSSION THREAD

**Discussion by John-Rocha** [Flutterando/forum#44] [Perguntas e Respostas] [answered]
Bom dia galera.

Eu estou com uma dúvida com relação à declaração de um valor default em um atributo de uma class.

Tenho a seguinte class: 

```dart
class WorkerCommunication {
  final bool? agree;
  DateTime? agreeDate;

  WorkerCommunication({
    this.agree,
    this.agreeDate,
  });
}
```
Daí eu preciso que o atributo `agreeDate` venha com `DateTime.now()` como default toda vez que a class for estanciada.

Alguma sugestão?

**Comment by DouglasValerio** [comment, marked as answer]
Opa,
Creio que você possa fazer dessa forma aqui:

```dart
class WorkerCommunication {
  final bool? agree;
  DateTime? agreeDate;

  WorkerCommunication({
    this.agree,
    DateTime? agreeDate
  }): agreeDate = agreeDate ?? DateTime.now();
}
```

**Reply by John-Rocha** [reply]
Opa, eu fiz exatamente desse jeito

**Reply by brunosemfio** [reply]
@John-Rocha marca a resposta do colega como aceita.
